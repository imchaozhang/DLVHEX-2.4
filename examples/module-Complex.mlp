#module(main,[]).
q(a).
q(b).
q(c).
callNext :- @p1[q]::callNext.

#module(p1,[q1/1]).
q1i(X) v q1i(Y) :- q1(X), q1(Y), X!=Y.
callP2 :- @p2[q1i]::callNext2.
callNext :- @p1[q1i]::callNext.

#module(p2,[q2/1]).
q2i(X) v q2i(Y) :- q2(X), q2(Y), X!=Y.
callNext2 :- @p2[q2i]::callNext2.


