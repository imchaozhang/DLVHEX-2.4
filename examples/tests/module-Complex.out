(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(b),q1i(c)}, p2[{q2(b),q2(c)}]={q2(b),q2(c),q2i(b)}, p2[{q2(b)}]={q2(b)}, p1[{q1(b),q1(c)}]={q1(b),q1(c),q1i(b)}, p1[{q1(b)}]={q1(b)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(b),q1i(c)}, p2[{q2(b),q2(c)}]={q2(b),q2(c),q2i(b)}, p2[{q2(b)}]={q2(b)}, p1[{q1(b),q1(c)}]={q1(b),q1(c),q1i(c)}, p2[{q2(c)}]={q2(c)}, p1[{q1(c)}]={q1(c)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(b),q1i(c)}, p2[{q2(b),q2(c)}]={q2(b),q2(c),q2i(c)}, p2[{q2(c)}]={q2(c)}, p1[{q1(b),q1(c)}]={q1(b),q1(c),q1i(b)}, p2[{q2(b)}]={q2(b)}, p1[{q1(b)}]={q1(b)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(b),q1i(c)}, p2[{q2(b),q2(c)}]={q2(b),q2(c),q2i(c)}, p2[{q2(c)}]={q2(c)}, p1[{q1(b),q1(c)}]={q1(b),q1(c),q1i(c)}, p1[{q1(c)}]={q1(c)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(c)}, p2[{q2(a),q2(c)}]={q2(a),q2(c),q2i(a)}, p2[{q2(a)}]={q2(a)}, p1[{q1(a),q1(c)}]={q1(a),q1(c),q1i(a)}, p1[{q1(a)}]={q1(a)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(c)}, p2[{q2(a),q2(c)}]={q2(a),q2(c),q2i(a)}, p2[{q2(a)}]={q2(a)}, p1[{q1(a),q1(c)}]={q1(a),q1(c),q1i(c)}, p2[{q2(c)}]={q2(c)}, p1[{q1(c)}]={q1(c)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(c)}, p2[{q2(a),q2(c)}]={q2(a),q2(c),q2i(c)}, p2[{q2(c)}]={q2(c)}, p1[{q1(a),q1(c)}]={q1(a),q1(c),q1i(a)}, p2[{q2(a)}]={q2(a)}, p1[{q1(a)}]={q1(a)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(c)}, p2[{q2(a),q2(c)}]={q2(a),q2(c),q2i(c)}, p2[{q2(c)}]={q2(c)}, p1[{q1(a),q1(c)}]={q1(a),q1(c),q1i(c)}, p1[{q1(c)}]={q1(c)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(b)}, p2[{q2(a),q2(b)}]={q2(a),q2(b),q2i(a)}, p2[{q2(a)}]={q2(a)}, p1[{q1(a),q1(b)}]={q1(a),q1(b),q1i(a)}, p1[{q1(a)}]={q1(a)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(b)}, p2[{q2(a),q2(b)}]={q2(a),q2(b),q2i(a)}, p2[{q2(a)}]={q2(a)}, p1[{q1(a),q1(b)}]={q1(a),q1(b),q1i(b)}, p2[{q2(b)}]={q2(b)}, p1[{q1(b)}]={q1(b)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(b)}, p2[{q2(a),q2(b)}]={q2(a),q2(b),q2i(b)}, p2[{q2(b)}]={q2(b)}, p1[{q1(a),q1(b)}]={q1(a),q1(b),q1i(a)}, p2[{q2(a)}]={q2(a)}, p1[{q1(a)}]={q1(a)})
(main[{}]={q(a),q(b),q(c)}, p1[{q1(a),q1(b),q1(c)}]={q1(a),q1(b),q1(c),q1i(a),q1i(b)}, p2[{q2(a),q2(b)}]={q2(a),q2(b),q2i(b)}, p2[{q2(b)}]={q2(b)}, p1[{q1(a),q1(b)}]={q1(a),q1(b),q1i(b)}, p1[{q1(b)}]={q1(b)})
